RecView.cpp

```diff
@@ -120,15 +116,29 @@ void RecView::OnPrint(CDC* pDC, CPrintInfo* pInfo)
 		pDC->TextOut(pageWidth / 4, yPos, rs.m_name);
 		if (rs.m_manager) {
 			manager = _T("x");
+			pDC->TextOut((pageWidth / 2), yPos, manager);
 		}
-		pDC->TextOut(2 * (pageWidth / 4), yPos, manager);
-
-		yPos += borderHeight;
-		pDC->MoveTo(leftMargin, yPos);
-		pDC->LineTo(rightMargin + 30, yPos);
 		
-		yPos += lineHeight;
 		rs.MoveNext();
+		if(!rs.IsEOF()) 
+		{
+			pDC->SelectObject(pen);
+			yPos += borderHeight;
+			pDC->MoveTo(leftMargin, yPos);
+			pDC->LineTo(rightMargin, yPos);
+		
+			yPos += lineHeight;
+		}
+		else 
+		{
+			pDC->SelectObject(pOldPen);

```

Comment:
Zašto if/else (sa uvjetom istim kao petlja) unutar petlje, zašto ovaj dio nije jednostavno iza petlje?

Reply:
Zato što nisam htio da mi ispiše isprekidanu liniju iza zadnjeg recorda, a uvjet je isti jer mi se činilo kao najjednostavnija provjera kraja baze. Zato sam stavio da je rc.MoveNext(); prije if da može u else a nakon toga i ovako i onako izađe iz petlje.